### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by maciejlotysz**

### CODE REVIEW

File: Cinema Room Manager/task/src/cinema/Menu.java [hunk 1]
```diff
@@ -26,7 +26,11 @@ public void displayMenu() {
                     System.out.println(roomView);
                     break;
                 case 2:
-                    room.buyTicket();
+                    System.out.println("Enter a row number:");
```
Comment (bogdanpolak) [inline Cinema Room Manager/task/src/cinema/Menu.java]: refactoring should not change logic of the code and you changed it.
previously there was a loop with validation and some hints for user to understand why inputed data are incorrect and user was able to buy only valid ticket.

you should validate seat position inside buy ticket method and rise an exception - this is the most popular solution solution

good alternative is to use Result monad. Read more here:
https://medium.com/@afcastano/monads-for-java-developers-part-2-the-result-and-log-monads-a9ecc0f231bb

Comment (bogdanpolak) [reply Cinema Room Manager/task/src/cinema/Menu.java]: Simple solution is to create to validation methods in Room class:
* isSeatValid(rowNum, seatNum)
* isSeatAvailable(rowNum, seatNum)

File: Cinema Room Manager/task/src/cinema/Room.java [hunk 2]
```diff
@@ -112,8 +111,8 @@ public String getSeatsView() {
         return sb.toString();
     }
 
-    public void buyTicket() {
-        selectSeat();
+    public void buyTicket(int rowNum, int seatNum) {
+        selectSeat(rowNum, seatNum);
```
Comment (bogdanpolak) [inline Cinema Room Manager/task/src/cinema/Room.java]: Call `seat_sell(rowNum, seatNum)` directly here and remove validation code as I was mentioned above

File: Cinema Room Manager/task/src/cinema/Room.java [hunk 3]
```diff
@@ -112,8 +111,8 @@ public String getSeatsView() {
...
         settingTicketPrice();
```
Comment (bogdanpolak) [inline Cinema Room Manager/task/src/cinema/Room.java:115]: add rowNum param to the `settingTicketPrice ` method

File: Cinema Room Manager/task/src/cinema/Room.java [hunk 4]
```diff
@@ -112,8 +111,8 @@ public String getSeatsView() {
...
         countCurrentIncome();
```
Comment (bogdanpolak) [inline Cinema Room Manager/task/src/cinema/Room.java:116]: add rowNum param to the `countCurrentIncome` method

File: Cinema Room Manager/task/src/cinema/Menu.java (hunk 1 above)
Comment (bogdanpolak) [reply Cinema Room Manager/task/src/cinema/Menu.java]:
```java
        bool seatEntered = false;
        while (!seatEntered) {
            System.out.println("Enter a row number:");
            selectedRowNumber = sc.nextInt();
            System.out.println("Enter a seat number in that row:");
            selectedSeatNumber = sc.nextInt();
            if (!room.isSeatValid(selectedRowNumber, selectedSeatNumber)) {
                System.out.println("Wrong input");
            } else if (!room.isSeatAvailable(selectedRowNumber, selectedSeatNumber)) {
                System.out.println("That ticket has already been purchased!");
            } else {
                seatEntered = true;
            }
        };
        room.buyTicket(selectedRowNumber, selectedSeatNumber);
```

Comment (maciejlotysz) [reply Cinema Room Manager/task/src/cinema/Menu.java]: I've set seat validation in class Menu.. Logic of the code is now restore.

File: Cinema Room Manager/task/src/cinema/Room.java (hunk 2 above)
Comment (maciejlotysz) [reply Cinema Room Manager/task/src/cinema/Room.java]: Done.

**Pull request merged**